hcl/eval/eval.go

```diff
@@ -111,12 +108,44 @@ func (c *Context) PartialEval(expr hclsyntax.Expression) (hclwrite.Tokens, error
 	return engine.Eval()
 }
 
-// GetExpressionTokens gets the provided expression writable tokens.
-func GetExpressionTokens(hcldoc []byte, expr hclsyntax.Expression) (hclwrite.Tokens, error) {
-	exprRange := expr.Range()
+// TokensForValue returns the tokens for the provided value.
+func TokensForValue(value cty.Value) (hclwrite.Tokens, error) {
+	value, marks := value.Unmark()
+	if value.Type() == customdecode.ExpressionClosureType {
+		closureExpr := value.EncapsulatedValue().(*customdecode.ExpressionClosure)
```

Comment:
```suggestion
	if closureExpr, ok := value.EncapsulatedValue().(*customdecode.ExpressionClosure); ok {
```

Not sure if it is better, feel free to ignore.

Reply:
as we have the handy `value.Type()` then we can avoid a (probably) complex runtime code path, most of the time, because encapsulated types are only used with ternary.

Reply:
hmm not sure the check for type on a interface is very complex, I think it simply checks the type on the interface internal struct, but I get it :+1: